tests/foreman/api/test_permission.py

```diff
@@ -41,17 +63,17 @@ def test_search_by_name(self, permission_name):
         is the one searched for.
 
         """
+        permission_name = random.choice(self.permission_names)
         if (
                 permission_name == 'power_compute_resources_vms' and
                 bz_bug_is_open(1198731)):
-            self.skipTest('BZ 1198731 is open.')
+            permission_name = random.choice(self.permission_names)
```

Comment:
Is there any particular reason to test only one random permission name? The process of searching for all permission names is what let us discover BZ 1198731, and if we only check one permission name, then future issues like that can crop up without us knowing. How about something like this:

```python
for permission_name in permission_names:
    if (
            permission_name == 'power_compute_resources_vms' and
            bz_bug_is_open(1198731)):
        continue
    result = entities.Permission(name=permission_name).search()
    self.assertEqual(len(result), 1, permission_name)
    self.assertEqual(permission_name, result[0][['name'])
```

And if we were using Python 3:

```python
for permission_name in permission_names:
    if (
            permission_name == 'power_compute_resources_vms' and
            bz_bug_is_open(1198731)):
        continue
    with self.subTest(permission_name=permission_name)
        result = entities.Permission(name=permission_name).search()
        self.assertEqual(len(result), 1)
        self.assertEqual(permission_name, result[0][['name'])
```